xbmc/interfaces/legacy/InfoTagVideo.cpp

```diff
@@ -176,5 +176,25 @@ namespace XBMCAddon
     {
       return infoTag->m_strTrailer;
     }
+
+    String InfoTagVideo::getArtist()
+    {
+      return StringUtils::Join(infoTag->m_artist, g_advancedSettings.m_videoItemSeparator);
```

Comment:
APIs shouldn't depend on user settings. Just return the original string list such that the consumers can format it as they please

Reply:
Makes sense! This is done, see c1ebe41efa8c2f1aaaea7a7bd06fad57dbb1b25b.

Note: _getDirector()_, _getWritingCredits()_ and _getGenre()_ also do this. I dare not change them for fear of breaking existing functionality (this being said I'd love to, but I feel I would need some guidance).

Reply:
Perhaps you could add new API methods getDirectors(), getWriters() and getGenres() which return a proper list, and we deprecate the old ones then? (this should be done in a separate PR though)
Or we delay bw-incompatible  API changes to the point when we switch to Py3?

Reply:
No we don't want to break Python API during the switch to Py3.